src/control/server/engine/config.go

```diff
@@ -18,23 +19,122 @@ import (
 )
 
 const (
-	maxHelperStreamCount = 2
-	envLogMasks          = "D_LOG_MASK"
-	envLogDbgStreams     = "DD_MASK"
-	envLogSubsystems     = "DD_SUBSYS"
+	maxHelperStreamCount         = 2
+	numPrimaryProviders          = 1
+	defaultNumSecondaryEndpoints = 1
+
+	// MultiProviderSeparator delineates between providers in a multi-provider config.
+	MultiProviderSeparator = ","
+
+	envLogMasks      = "D_LOG_MASK"
+	envLogDbgStreams = "DD_MASK"
+	envLogSubsystems = "DD_SUBSYS"
 )
 
 // FabricConfig encapsulates networking fabric configuration.
 type FabricConfig struct {
-	Provider        string `yaml:"provider,omitempty" cmdEnv:"CRT_PHY_ADDR_STR"`
-	Interface       string `yaml:"fabric_iface,omitempty" cmdEnv:"OFI_INTERFACE"`
-	InterfacePort   int    `yaml:"fabric_iface_port,omitempty" cmdEnv:"OFI_PORT,nonzero"`
-	NumaNodeIndex   uint   `yaml:"-"`
-	BypassHealthChk *bool  `yaml:"bypass_health_chk,omitempty" cmdLongFlag:"--bypass_health_chk" cmdShortFlag:"-b"`
-	CrtCtxShareAddr uint32 `yaml:"crt_ctx_share_addr,omitempty" cmdEnv:"CRT_CTX_SHARE_ADDR"`
-	CrtTimeout      uint32 `yaml:"crt_timeout,omitempty" cmdEnv:"CRT_TIMEOUT"`
-	DisableSRX      bool   `yaml:"disable_srx,omitempty" cmdEnv:"FI_OFI_RXM_USE_SRX,invertBool,intBool"`
-	AuthKey         string `yaml:"fabric_auth_key,omitempty" cmdEnv:"D_PROVIDER_AUTH_KEY"`
+	Provider              string `yaml:"provider,omitempty" cmdEnv:"CRT_PHY_ADDR_STR"`
+	Interface             string `yaml:"fabric_iface,omitempty" cmdEnv:"OFI_INTERFACE"`
+	InterfacePort         string `yaml:"fabric_iface_port,omitempty" cmdEnv:"OFI_PORT"`
```

Comment:
I recall this was all decided pretty early on, to shoehorn the secondary providers/interfaces/ports into the existing fields, rather than add new fields. I tend to agree in productizing this, it may be better to add `secondary_providers`, `secondary_fabric_ifaces`, etc.

Reply:
Yeah, IMO there's no reason that the engine configuration details have to be exposed to the user. You could add the new user-visible fields for secondary details and then combine everything behind the scenes into the existing engine config parameters.